cwms-data-api/src/main/java/cwms/cda/api/LevelsController.java

```diff
@@ -280,23 +278,14 @@ public void getAll(@NotNull Context ctx) {
 
             boolean isLegacyVersion = version.equals("1");
 
-            if (format.isEmpty() && !isLegacyVersion)
-            {
+            if (format.isEmpty() && !isLegacyVersion) {
                 String cursor = ctx.queryParamAsClass(PAGE, String.class)
                                    .getOrDefault("");
                 int pageSize = ctx.queryParamAsClass(PAGE_SIZE, Integer.class)
                                   .getOrDefault(DEFAULT_PAGE_SIZE);
 
-                ZoneId tz = ZoneId.of(timezone, ZoneId.SHORT_IDS);
-
-                ZonedDateTime endZdt = end != null ? DateUtils.parseUserDate(end, timezone) :
-                        ZonedDateTime.now(tz);
-                ZonedDateTime beginZdt;
-                if (begin != null) {
-                    beginZdt = DateUtils.parseUserDate(begin, timezone);
-                } else {
-                    beginZdt = endZdt.minusHours(24);
-                }
+                ZonedDateTime endZdt = end != null ? DateUtils.parseUserDate(end, timezone) : null;
```

Comment:
use `requiredInstant(ctx, END);` or `requiredZdt(ctx, END);` (not sure why this time needs to be zoned).

Reply:
The beginning and end parameters are not required fields. I replaced the ternaries with `queryParamAsZdt` instead.